File: 프로그래머스/weeklyChallenge/002.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string solution(vector<vector<int>> scores) {
    string answer = "";
    
    for(int i=0;i<scores.size();i++){
        int max = -1;
        int min = 101;
        int max_check = 0;
        int min_check = 0;
        int sum = 0;
        int num = scores.size();
        float average = 0;
        string grade = "";
        for(int j=0;j<scores.size();j++){
            if(scores[j][i] > max){
                max = scores[j][i];
                max_check = 0;
            }else if(scores[j][i] == max){
                max_check ++;
            }
            if(scores[j][i] < min){
                min = scores[j][i];
                min_check = 0;
            }else if(scores[j][i] == min){
                min_check ++;
            }
            sum = sum + scores[j][i];
        }
        if(max == scores[i][i] &&  max_check == 0){
            sum = sum - max;
            num --;
        }else if(min == scores[i][i] && min_check == 0){
            sum = sum - min;
            num --;
        }
        average = (float)sum / (float)num;
        /*
        cout << "max : " << max << ", " << max_check << "\n";
        cout << "min : " << min << ", " << min_check << "\n";
        cout << "sum : " << sum << ", " << average << "\n";
        cout << "\n";
        */
        if(average >= 90){
            grade = "A";
        }else if(average >= 80){
            grade = "B";
        }else if(average >= 70){
            grade = "C";
        }else if(average >= 50){
            grade = "D";
        }else{
            grade = "F";
        }
        answer = answer + grade;
    }
    return answer;
}

File: 프로그래머스/weeklyChallenge/002.cs
using System;
using System.Collections.Generic;

public class Solution {
    public string solution(int[,] scores) {
        string answer = "";
        var s = new List<int>();
        var v = new List<float>();
        //List<int> s;
        for(int i=0;i<scores.GetLength(0);i++){
            int max = -1;
            bool maxFlag = false;
            int min = 101;
            bool minFlag = false;
            int sum = 0;
            int num = scores.GetLength(1);
            for(int j=0;j<scores.GetLength(1);j++){
                if(scores[j,i] > max){
                    max = scores[j,i];
                    maxFlag = false;
                }else if(scores[j,i] == max){
                    maxFlag = true;
                }
                
                if(scores[j,i] < min){
                    min = scores[j,i];
                    minFlag = false;
                }else if(scores[j,i] == min){
                    minFlag = true;
                }
                sum += scores[j,i];
            }
            if(!maxFlag && max == scores[i,i]){
                sum -= scores[i,i];
                num --;
            }else if(!minFlag && min == scores[i,i]){
                sum -= scores[i,i];
                num --;
            }
            v.Add((float)sum / (float)num);
        }
        foreach(float e in v){
            //Console.Write(e+" ");
            char g;
            if(e >= 90){
                g = 'A';
            }else if(e >= 80){
                g = 'B';
            }else if(e >= 70){
                g = 'C';
            }else if(e >= 50){
                g = 'D';
            }else{
                g = 'F';
            }
            answer += g;
        }
        //Console.WriteLine(scores.GetLength(1));
        return answer;
    }
}